### DISCUSSION TITLE
Implementing React.js in Pimcore docs?

### DISCUSSION THREAD

**Discussion by Dizzyggs** [pimcore/pimcore#12857] [Q&A / Support] [answered]
Hello everyone. Im new to Pimcore so please bare with me.

I was just curious on how to implement React.js in pimcore? I can't find any docs on how to do it, and no tutorials out there either.

Thank you!

**Comment by ChrisB9** [comment, marked as answer]
Are you planning on using React.js in your frontend or backend?

**Reply by Dizzyggs** [reply]
Hello, Frontend specifically!

**Reply by ChrisB9** [reply]
Then that's nothing pimcore related tbh. 
The most efficient and symfonian way is through this: https://symfony.com/doc/current/frontend.html
Comes with direct support for twig and react, so you just have to add dependencies to your project and start coding. I am using it that way too, and it works very very well :)

One pimcore-thing though:
If you decide to use pimcore headless, then your best bet is this: https://github.com/pimcore/data-hub. That extensions handles everything you desire in pimcore with GraphQL and works wonderful too

**Reply by Dizzyggs** [reply]
Thank you @ChrisB9 for the tips, and sry for the late response.

Will try later this week, thanks for the headsup! :)
